Source/Dafny/Coco/Scanner.frame

```diff
@@ -453,13 +463,53 @@ public class Scanner {
 
   Token/*!*/ NextToken() {
     Contract.Ensures(Contract.Result<Token>() != null);
+    if(!parsingTrivia) {
+      posTrailingStart = pos;
+      posTrailingEndLeadingStart = pos;
+      parsingTrivia = true;
+      triviaTrailing = pos > 0;
+      triviaNewlines = 0;
+      lastWasCr = false;
+    } else {
+      // We just skipped a comment. Any newline coming after the comment doesn't count for the splitting of the trivia
+      if(triviaTrailing && (ch == '\r' || (ch == '\n' && !lastWasCr))) {
+        triviaNewlines--;
+        lastWasCr = (ch == '\r');
+      }
+    }
     while (ch == ' ' ||
 -->scan1
-    ) NextCh();
+    ) {
+      if(triviaTrailing && (ch == '\r' || ch == '\n')) {
+        if(ch == '\r' || (ch == '\n' && !lastWasCr)) { // Don't count \r\n as two newlines.
+          triviaNewlines++;
```

Comment:
What's the definition of trailing trivia? Shouldn't any newlines interrupt set `triviaTrailing = false;` ?